### ISSUE TITLE
Unable to install on Windows 2012 R2 (unconfirmed)

### ISSUE THREAD

**Issue by mauricel** [fredrikaverpil/vcpython27#1]
Hi Fedrik, 
Just wanted to let you know that I tried your package, but  I'm not able to get your new package to work. Perhaps it's something I'm doing? 

Environment: Windows Server 2012 R2, running as Administrator

git clone git@github.com:fredrikaverpil/vcpython27.git
cd vcpython27
choco pack
mkdir ..\test
move *.nupkg  ..\test
cd ..
choco install vcpython27 -s test

I get the following log:

[chocolatey.txt](https://github.com/fredrikaverpil/vcpython27/files/122385/chocolatey.txt)

No need to do anything just yet, I'll try again in the morning on a fresh VM. Just wanted to keep you in the loop.

Thanks for the patch!

**Comment by fredrikaverpil** [comment]
I was using the instructions from here: https://github.com/chocolatey/choco/wiki/CreatePackages#testing-your-package

Since the dependencies are not in my source, I appended `-source "'%cd%;https://chocolatey.org/api/v2/'"` and then it installed without errors.

**Comment by fredrikaverpil** [comment]
Ah, yes, I'm sorry. I do get an error. Hm, I wonder why it worked yesterday.

It looks good up until all downloads are complete:

	C:\Users\fredrik\Documents\GitHub>choco install vcpython27 -s test -source "'%cd%;https://chocolatey.org/api/v2/'"
	Installing the following packages:
	vcpython27
	By installing you accept licenses for the packages.

	vcredist2008 v9.0.30729.6161
	The package vcredist2008 wants to run 'chocolateyInstall.ps1'.
	Note: If you don't run this script, the installation will fail.
	Note: To confirm automatically next time, use '-y' or consider setting
	 'allowGlobalConfirmation'. Run 'choco feature -h' for more details.
	Do you want to run the script?
	 1) yes
	 2) no
	 3) print
	1
	 Downloading vcredist2008 64 bit
	   from 'https://download.microsoft.com/download/5/D/8/5D8C65CB-C849-4025-8E95-C3966CAFD8AE/vcredist_x64.exe'
	 Hashes match.
	 Installing vcredist2008...
	 vcredist2008 has been installed.
	 Downloading vcredist2008 32 bit
	   from 'https://download.microsoft.com/download/5/D/8/5D8C65CB-C849-4025-8E95-C3966CAFD8AE/vcredist_x86.exe'
	 Hashes match.
	 Installing vcredist2008...
	 vcredist2008 has been installed.
	 The install of vcredist2008 was successful.

	DotNet3.5 v3.5.20160116
	The package DotNet3.5 wants to run 'ChocolateyInstall.ps1'.
	Note: If you don't run this script, the installation will fail.
	Note: To confirm automatically next time, use '-y' or consider setting
	 'allowGlobalConfirmation'. Run 'choco feature -h' for more details.
	Do you want to run the script?
	 1) yes
	 2) no
	 3) print
	1
	 Microsoft .Net 3.5 Framework is already installed on your machine.
	 The install of dotnet3.5 was successful.

	vcpython27 v9.0.0.30729
	The package vcpython27 wants to run 'chocolateyinstall.ps1'.
	Note: If you don't run this script, the installation will fail.
	Note: To confirm automatically next time, use '-y' or consider setting
	 'allowGlobalConfirmation'. Run 'choco feature -h' for more details.
	Do you want to run the script?
	 1) yes
	 2) no
	 3) print
	1

This is where the downloaded files are not found:

	 Copying vcpython27
	   from '[-validExitCodes'
	 Chocolatey expected a file to be downloaded to 'C:\Users\fredrik\AppData\Local\Temp\chocolatey\vcpython27\9.0.0.30729\v
	 cpython27Install.' but nothing exists at that location.
	 At C:\ProgramData\chocolatey\helpers\functions\Get-ChocolateyWebFile.ps1:158 char:24
	 + ... .Exists)) { throw "Chocolatey expected a file to be downloaded to `'$ ...
	 +                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
	     + CategoryInfo          : OperationStopped: (Chocolatey expe... that location.:String) [], RuntimeException
	     + FullyQualifiedErrorId : Chocolatey expected a file to be downloaded to 'C:\Users\fredrik\AppData\Local\Temp\choc 
	    olatey\vcpython27\9.0.0.30729\vcpython27Install.' but nothing exists at that location.
	The install of vcpython27 was NOT successful.
	Error while running 'C:\ProgramData\chocolatey\lib\vcpython27\tools\chocolateyinstall.ps1'.
	 See log for details.

	Chocolatey installed 2/3 package(s). 1 package(s) failed.
	 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
	Failures:
	 - vcpython27

I'll try to take a look at this tomorrow.

**Comment by fredrikaverpil** [comment]
It works when being run in a cmd.exe shell on Windows 10 Pro 64-bit:

	C:\nupkgs>choco install vcpython27 -s test -source "'%cd;https://chocolatey.org/api/v2'"
	Installing the following packages:
	vcpython27
	By installing you accept licenses for the packages.

	vcpython27 v0.1 (forced)
	The package vcpython27 wants to run 'chocolateyInstall.ps1'.
	Note: If you don't run this script, the installation will fail.
	Note: To confirm automatically next time, use '-y' or consider setting
	 'allowGlobalConfirmation'. Run 'choco feature -h' for more details.
	Do you want to run the script?
	 1) yes
	 2) no
	 3) print
	1
	 Installing vcpython27...
	 vcpython27 has been installed.
	 The install of vcpython27 was successful.

	Chocolatey installed 1/1 package(s). 0 package(s) failed.
	 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

The file is downloaded into `C:\Users\iruser\AppData\Local\Temp\chocolatey\vcpython27\0.1` for me, which is equal to `$env:temp\chocolatey\vcpython27\0.1`.  I wonder exactly where in the [chocolateyInstall.ps1](https://github.com/fredrikaverpil/vcpython27/blob/master/tools/chocolateyinstall.ps1) this directory is being defined. I can only see it being present in the command argument (the `silentArgs` variable).

Could you please try again and verify what's happening in your TEMP folder prior, during and after installation?

**Comment by fredrikaverpil** [comment]
I updated the post above, so in case you already read it ... please read again :)

**Comment by fredrikaverpil** [comment]
Sorry, my `Install-ChocolateyPackage` was malformed. I've reproduced your initial exact error and updated the code: https://github.com/fredrikaverpil/vcpython27/commit/53fe2473132d939994e90ee47a477a93e3ce3a89

However, I'm now trying to [figure out](https://github.com/fredrikaverpil/vcpython27/issues/2) why "Microsoft Visual C++ Compiler Package for Python 2.7" doesn't appear in "Programs and Features". I've noticed this doesn't happen with the currently published 0.1 version of vcpython either. Unfortunately, this somehow makes it impossible to uninstall vcpython27 properly. I'd like to try and fix this.
